<?xml version="1.0"?>

<!--|
    Copyright (C) 2019 Learning Algorithms and Systems Laboratory, EPFL, Switzerland
    Authors:  Harshit Khurana, Maxime Gautier
    email:    harshit.khurana@epfl.ch, maxime.gautier@epfl.ch
    website:  lasa.epfl.ch

    This file is part of i_am_project EPFL (impact aware manipulation).
|-->

<launch>

  <param name="simulation_referential" type="bool" value="False"/>

  <include file="$(find iiwa_toolkit)/launch/passive_track_gazebo_dual.launch">
    <arg name="robot_name_1" value="iiwa1"/>
    <arg name="iiwa_model_1" value="7"/>

    <arg name="robot_name_2" value="iiwa2"/>
    <arg name="iiwa_model_2" value="14"/>

    <arg name="controller" value="TorqueController"/>
    
    <arg name="position_1" value="-x -0.0 -y -0.6"/>
    <arg name="position_2" value="-x -0.0 -y 0.6"/>
  </include>


  <!-- Environnment -->
  <node name="spawn_big_box" pkg="gazebo_ros" type="spawn_model" 
  args="-file $(find i_am_project)/object_models/big_box.urdf -x 0.55 -y 0.0 -z 0.025 -urdf -model table_model" />

  <node name="spawn_box" pkg="gazebo_ros" type="spawn_model" 
  args="-file $(find i_am_project)/object_models/box.urdf -x 0.55 -y -0.4 -z 0.175 -urdf -model box_model" />


  <!-- vrpn Optitrack -->
  <arg name="server" default="128.178.145.172"/>
  <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">
      <rosparam subst_value="true">
      server: $(arg server)
      port: 3883
      frame_id: world
      broadcast_tf: true
      # Must either specify refresh frequency > 0.0, or a list of trackers to create
      refresh_tracker_frequency: 1.0
      #trackers:
      #- FirstTracker
      #- SecondTracker
      </rosparam>
  </node>

  <!-- All ROS topics names -->
  <rosparam file="$(find i_am_project)/config/ros_topics_air_hockey.yaml" />
  <rosparam file="$(find i_am_project)/config/hit_properties_air_hockey.yaml" />
  <rosparam file="$(find i_am_project)/config/recorder_properties.yaml" />

  <!-- Here is the node that allows me to hit the object in the iiwa simulator -->
  <node pkg = "i_am_project" type = "air_hockey"  name = "air_hockey" output="screen"/>

</launch>

